### PR TITLE
[C#] Fix Struct Tests to build out of the box.

### LINKED ISSUE #1846: [C#] structs - ensure exercise compiles when first downloaded.
Currently we ask the student to implement the `Plot` struct in this exercise.  Because of this the code provided by the CLI (Structs.cs, Structs.csproj, StructsTests.cs) will not compile.

Please replace calls to `new Plot(...)` with a `CreatePlot()` method and use reflection to call the constructor if an appropriate one is present otherwise throw an `InvalidOperationException`.

This should enable the code to build "out of the box".

### PR DESCRIPTION
**Pull request by aes421**
Used reflection to invoke plot constructor or throw error if not found.  This allows the code to build out of the box instead of having build errors

Fixes #1846 

Maintainers - can you add the hacktoberfest-accepted label to my PR?  Thank you

### CODE REVIEW

**Comment by aes421** [comment]
@ErikSchierboom @mikedamay Any idea what's going on with the whitespace here?  I feel like I'm guessing in the dark haha

File: languages/csharp/exercises/concept/structs/StructsTests.cs [hunk 1]
```diff
@@ -7,48 +8,64 @@ public class StructsTests
...
+        Type[] types = new Type[] { typeof(Coord), typeof(Coord), typeof(Coord), typeof(Coord) };
+        ConstructorInfo constructorInfoObj = plotType.GetConstructor(types);
+        if (constructorInfoObj != null)
+        {
+            return (Plot)constructorInfoObj.Invoke(new object[] { coord1, coord2, coord3, coord4 });
+        }
+
+        else
+        {
+            throw new InvalidOperationException("Cannot find constructor that takes 4 coordinates");
```
Comment (mikedamay) [inline languages/csharp/exercises/concept/structs/StructsTests.cs:68]:
```suggestion
            throw new InvalidOperationException("You need to implement a constructor for the struct Plot.  The constructor must take 4 co-ordinates.  No such constructor can be found.");
```
I would suggest something like this.  I think it is equally likely that the student has omitted to implement any constructor as that they chose the wrong number of arguments.

**Comment by mikedamay** [comment]
This is great.

Thank you for picking it up.

One small niggle on the "failure" message.

**Review by mikedamay — APPROVED** [review]
Thank you.

Best of luck.  I hope you give us some consideration in the other 11 months of the year

**Comment by ErikSchierboom** [comment]
Thanks @aes421!

**Pull request merged**